### ISSUE TITLE
[BUG] /lol/match/v5/matches/{match_id} custom games with known match ID no longer obtainable

### ISSUE THREAD

**Issue by LukePeltier** [RiotGames/developer-relations#472]
In the previous version, if you knew the custom match ID, you could call `/lol/match/v4/matches/{matchId}` and see the match details of a completed custom game (just like how it is visible on matchhistory.na.leagueoflegends.com)

in v5, this match id (prepended by the region code) now returns a 404

Example custom game:
match id: 3833894399

https://americas.api.riotgames.com/lol/match/v5/matches/NA1_3833894399
404 Not Found

https://na1.api.riotgames.com/lol/match/v4/matches/3833894399
full match details

https://matchhistory.na.leagueoflegends.com/en/#match-details/NA1/3833894399?tab=overview
full match details

I use custom games for a personal match history website that a group of us use to track statistics in our games, but with these changes, I wont be able to get any details about the match through the api

**Comment by heitorpilotto** [comment]
I got the same issue here:

Full match details: https://br1.api.riotgames.com/lol/match/v4/matches/2236441575

When using v5: https://americas.api.riotgames.com/lol/match/v5/matches/BR1_2236441575 - return 404 not found

Math history:
https://matchhistory.br.leagueoflegends.com/pt/#match-details/BR1/2236441575/200298930?tab=overview

Building here for the same purpose with my group of friends.

**Comment by ArjunBhushan** [comment]
Bump if possible.

**Comment by onpointbtw** [comment]
https://www.reddit.com/r/leagueoflegends/comments/pfalpb/turning_off_web_match_history/hb4cy0m/?context=3

I think this is intended, with the Match History Website also shutting down, Custom Games are meant to be private - except you provide OAuth Authorization.

Here is a link to the devtracker: https://devtrackers.gg/leagueoflegends/p/571444a8-turning-off-web-match-history

**Comment by LukePeltier** [comment]
Right, we just don't have the OAuth method for those games yet I don't believe. I was planning on leaving the ticket open until there was a way to get the data.

**Comment by ajoscram** [comment]
Anyone got any updates on this? Supper bummed out because [I built a Discord bot](https://github.com/ajoscram/Jupi-Zordon-3400/) for keeping track of mine and my friends' custom match stats and do some AI stuff, but now it's impossible for me to obtain custom match data. Previously, I could record player identities with the `spectator-v4` endpoint during the match, and then ask for the match stats via `match-v4` once the game was done. Using those two pieces of info I was able to link player identities to their champions and get the complete match info and player stats. It was scrappy but it worked.

@RiotTuxedo (or anyone else for that matter): **is it currently impossible to obtain custom match data?** Other than the tournaments API, obviously. It would be very nice if we could just decide whether or not a custom game is public prior to starting the match, via the in-game UI.

**Comment by iaace-NA** [comment]
@ajoscram download the .rofl and parse the JSON header
or
exploit some LCU endpoint

**Comment by ajoscram** [comment]
@iaace-NA LCU endpoints I can't do because that would require running a program on a Client. If I were to play every single match myself that wouldn't be an issue (since I'd just record them myself), but that's not guaranteed. I also don't feel like a Discord bot should require you to run another program to work. .rofls on the other hand... that might just work, but I gotta look into them.

Both solutions are sorta hacky though, it would be nice to not have to circumvent systems to get custom match data if the match participants are willing to expose it.

Thanks for your input though!

**Comment by geozukunft** [comment]
They are not supposed to be optainable with match-v5

**Comment by Ryo-higuchi** [comment]
I am facing the same problem.I want to get a custom timeline, is there any way?

**Comment by geozukunft** [comment]
Support for custom games got removed with the release of match-v5. You could however get custom games through the LCU

**Comment by Ryo-higuchi** [comment]
I have my LCU so it seems possible. thank you

**Comment by Ryo-higuchi** [comment]
Is it possible to use the LCU on a PC without a LoL client?

**Comment by Ryo-higuchi** [comment]
Is it possible to use the LCU on another PC without the LoL client?

**Comment by thomasmarton** [comment]
@Ryo-higuchi 
The current version LoL client is called the League Client Update (aka the LCU) therefor it's not possible to use the LCU without itself.

**Comment by Ryo-higuchi** [comment]
If you want to get a custom timeline using LCU. Is there a way other than using /lol-match-history/v1/game-timelines/?

**Comment by onpointbtw** [comment]
> Support for custom games got removed with the release of match-v5. You could however get custom games through the LCU

For anyone wondering:

The endpoint for this is "/lol-match-history/v1/games/<gameId>" with the gameId being just the ID, not e.g. "EUW1_<gameId>"
The return from this endpoint still follows the old matchv4 schema  
( https://web.archive.org/web/20210612165448/https://developer.riotgames.com/apis#match-v4/GET_getMatch )

**Comment by McDoMaster** [comment]
Hi everyone,
I'm in a similar case with ajoscram but I'm trying to move on the LCU. 

But I have trouble obtaining the data and the following endpoints return 404 codes:

- /lol-match-history/v2/matchlist 
- /lol-match-history/v1/matchlist

Is the LCU stopped the use of those endpoints ? Anyone got some documentation that I could use ?
Thanks for your help

**Comment by geozukunft** [comment]
The LCU is only supported on a do at your own risk for changes basis. Best way to get an answer is to join the 3rd party dev discord and ask in #lcu-dev

**Comment by Resinderate** [comment]
> @ajoscram download the .rofl and parse the JSON header or exploit some LCU endpoint

I've been using the `.rofl` file approach here for a while, but looks like this data has been removed from `.rofl` files in patch 13.20 🙁

**Comment by xoxodlf** [comment]
Could you let me know when you plan to resolve it?

**Comment by geozukunft** [comment]
This feature had been first talked about in February 2021 so don't get your hopes up high to see anything in the foreseeable future @xoxodlf

**Comment by asivaneswaran** [comment]
So what's the use of Tournament API then? If we can't pull the game data, we can't pull the game data, so we can't automate on the winning team no?

**Comment by DarkIntaqt** [comment]
You can pull custom matches using oauth, also as far as I know you can pull tournament games with the tournament code in match v5

**Comment by bangingheads** [comment]
Custom games can now be fetched with [https://developer.riotgames.com/apis#lol-rso-match-v1](https://developer.riotgames.com/apis#lol-rso-match-v1)

Tournament API games can be fetched normally through Match-V5, using the callback, getting the game id through [https://developer.riotgames.com/apis#tournament-v5/GET_getGames](https://developer.riotgames.com/apis#tournament-v5/GET_getGames) or filtering a player's match history for the tourney type.

**Comment by lucasMesquitaBorges** [comment]
Thanks @bangingheads for the update.

**Comment by minhquan96321** [comment]
I have a 403 error while there is a key
